graphql/util.go

```diff
@@ -383,3 +385,84 @@ func ConvertDBTasksToGqlTasks(tasks []task.Task, baseTaskStatuses BaseTaskStatus
 	}
 	return taskResults
 }
+
+type Modifications struct {
+	Action   string   `json:"action"`
+	Active   bool     `json:"active"`
+	Abort    bool     `json:"abort"`
+	Priority int64    `json:"priority"`
+	TaskIds  []string `json:"task_ids"`
+}
+
+func ModifyVersion(version model.Version, user user.DBUser, proj *model.ProjectRef, modifications Modifications) (error, int) {
```

Comment:
Thanks for refactoring this, we desperately needed to separate logic from HTTP handling code 👍

Reply:
you got it dude 👍